sub/osd_libass.c

```diff
@@ -685,8 +685,14 @@ static void append_ass(struct ass_state *ass, struct mp_osd_res *res,
 struct sub_bitmaps *osd_object_get_bitmaps(struct osd_state *osd,
                                            struct osd_object *obj, int format)
 {
-    if (obj->type == OSDTYPE_OSD && obj->osd_changed)
-        update_osd(osd, obj);
+    if (obj->type == OSDTYPE_OSD) {
+        if (obj->osd_changed) {
```

Comment:
probably unreachable in practice but an `osd_state` is created with `osd_changed` set to false, so this would break. could simply init it to true.

Reply:
The osd is checked and compared to previous one always before this function happen. So I don't think it's a problem.

Unless there was preexisting bug, because the logic of `osd_changed` doesn't change in this PR.

Reply:
There is no bug, I am just proposing safeguards so that:
```c
auto *x = osd_create(mp_global);
osd_render(osd, ...);
```
doesn't crash (because `osd_changed` is false but `ass_packer` null

Reply:
I see, but such case would be invalid use of `OSDTYPE_OSD` overlay. The idea is to have "something", always check and initialize `osd_changed` as needed.

Initializing `osd_changed` to `true` initially would hide this issue and force osd to always render making `osd_changed` not functional, because no one would clear it to `false`.

At least I see it like that.